designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessActionRepository.scala

```diff
@@ -264,6 +264,22 @@ abstract class DbProcessActionRepository[F[_]](
     run(query.result.map(_.toSet))
   }
 
+  def getInProgressCanceledAndDeployed: F[Map[ProcessId, Set[ProcessActionType]]] = {
+    val query = processActionsTable
+      .filter(action =>
+        action.state === ProcessActionState.InProgress &&
+          action.actionType
+            .inSet(ProcessActionType.Deploy :: ProcessActionType.Cancel :: Nil)
```

Comment:
I like to have more general purpose methods in repository and move things that we can move upper without performance lost to services. I mean instead of having  `getInProgressCanceledAndDeployed` we can have `getInProgressActionTypes(allowedActionTypes: Set[ProcessActionType])`